app/services/journeyAnswers/NICsAnswersService.scala

```diff
@@ -58,17 +57,84 @@ class NICsAnswersServiceImpl @Inject() (connector: IFSConnector, repository: Jou
       _ <- repository.upsertAnswers(ctx.toJourneyContext(JourneyName.NationalInsuranceContributions), Json.toJson(storageAnswers))
     } yield ()
 
-  def getAnswers(ctx: JourneyContextWithNino)(implicit hc: HeaderCarrier): ApiResultT[Option[NICsAnswers]] =
-    for {
-      apiAnswers <- connector.getDisclosuresSubmission(ctx)
-      dbAnswers  <- repository.getAnswers[NICsStorageAnswers](ctx.toJourneyContext(JourneyName.NationalInsuranceContributions))
-    } yield NICsAnswers.mkPriorClass2Data(apiAnswers, dbAnswers)
-
   private def upsertOrDeleteClass2Data(maybeClass2Nics: Option[RequestSchemaAPI1638], ctx: JourneyContextWithNino)(implicit
       hc: HeaderCarrier): ApiResultT[Unit] =
     maybeClass2Nics match {
       case Some(data) => connector.upsertDisclosuresSubmission(ctx, data).void
       case None       => connector.deleteDisclosuresSubmission(ctx)
     }
 
+  def saveClass4SingleBusiness(ctx: JourneyContextWithNino, answers: NICsClass4Answers)(implicit hc: HeaderCarrier): ApiResultT[Unit] =
+    for {
+      updatedContext  <- updateJourneyContextWithSingleBusinessId(ctx)
+      existingAnswers <- getBusinessAnnualSummaryOrMakeNew(updatedContext)
+      requestBody   = CreateAmendSEAnnualSubmissionRequestData.mkNicsClassFourSingleBusinessRequestBody(answers, existingAnswers)
+      upsertRequest = CreateAmendSEAnnualSubmissionRequestData(ctx.taxYear, ctx.nino, updatedContext.businessId, requestBody)
+      _ <- EitherT[Future, ServiceError, Unit](connector.createAmendSEAnnualSubmission(upsertRequest))
+    } yield ()
+
+  private def updateJourneyContextWithSingleBusinessId(ctx: JourneyContextWithNino)(implicit hc: HeaderCarrier): ApiResultT[JourneyContextWithNino] =
+    EitherT(businessConnector.getBusinesses(ctx.nino).value.map {
+      case Right(res: api_1171.SuccessResponseSchema) =>
+        res.taxPayerDisplayResponse.getMaybeSingleBusinessId match {
+          case Some(id) => Right(ctx(newId = id))
+          case None     => Left(BusinessNotFoundError(ctx.businessId))
+        }
+      case Left(error) => Left(error)
+    })
+
+  private def getBusinessAnnualSummaryOrMakeNew(updatedContext: JourneyContextWithNino)(implicit
+      hc: HeaderCarrier): ApiResultT[api_1803.SuccessResponseSchema] =
+    EitherT(connector.getAnnualSummaries(updatedContext).map {
+      case Left(error) if error.status == NOT_FOUND => Right(api_1803.SuccessResponseSchema.empty)
+      case otherResult                              => otherResult
+    })
+
```

Comment:
If there's a neater way to do updateJourneyContextWithSingleBusinessId or getBusinessAnnualSummaryOrMakeNew, please let me know